### PR TITLE
#2 Implement user registration flow

### LINKED ISSUE #2: Implement user registration flow
# Description

- As a trainee, I should be able to accept the invitation to join the organization and proceed with the registration flow
- Appropriate error should be displayed

### PR DESCRIPTION
**Pull request by fredshema**
### What does this PR do?

This PR includes design and implementation of trainee registration.

### Description of Task to be completed?

Once a trainee is invited in an organization, they receive an email with a link to register. Once they click on the link they will be redirected to the app on the registration page. The page receives a token and a user fills in their information. Once they submit, an account is created and they are redirected to the trainee dashboard.

### How should this be manually tested?

- Invite a user using the frontend and make sure they receive an invitation email (set smtp related environment variables to be able to send emails).
- On the testing device, open the invitation email and click on it, the app will be opened on the registration page.
- Fill in all required information and submit the form.

### CODE REVIEW

**Review by munezeromicha — APPROVED** [review]

File: app/auth/register.tsx [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+            {loading ? (
+              <ActivityIndicator color="#fff" />
+            ) : (
+              <Text className={`text-white text-lg`}>Register</Text>
+            )}
+          </TouchableOpacity>
+        </View>
+        <View className="mt-6">
+          <Text>
+            Already have an account? <Text className="text-action-500">Sign in</Text>
```
Comment (munezeromicha) [inline app/auth/register.tsx]: Here the user will be able to switch to sign in screen after pressing on sign in. if already have an account.

**Review by munezeromicha — CHANGES_REQUESTED** [review]

File: app/auth/register.tsx (hunk 1 above)
Comment (Mbabazi12) [reply app/auth/register.tsx]: Thank you for this, now it's good to go

**Review by munezeromicha — APPROVED** [review]

**Review by aimeudoxie — APPROVED** [review]
Good job

**Review by aimeudoxie — CHANGES_REQUESTED** [review]
Squash your commits

**Comment by fredshema** [comment]
> Squash your commits

@aimeudoxie commits have been squashed

**Review by Mbabazi12 — APPROVED** [review]

**Comment by ncutixavier** [comment]
Did the page's input not miss the labels? For instance, the first two inputs don't mean anything to me. @fredshema  & @Mbabazi12  can we have labels for those inputs

**Comment by Mbabazi12** [comment]
@ncutixavier I added the labels. Thank you!

**Review by aimeudoxie — APPROVED** [review]

File: app/auth/register.tsx [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+          ToastAndroid.show('Successfully registered', ToastAndroid.LONG);
+          await AsyncStorage.setItem('org_token', data.createUser.token);
+          router.push('/dashboard/trainee');
+        }
+
+        if (errors) {
+          ToastAndroid.show(errors[0].message, ToastAndroid.LONG);
+        }
+      } catch (error) {
+        console.log(error);
```
Comment (ncutixavier) [inline app/auth/register.tsx]: Remove all logs

**Review by ncutixavier — COMMENTED** [review]
Add screenshot for dark mode,

**Comment by ncutixavier** [comment]
Also, I can't see `.env.example` to setup env variables

File: app/auth/register.tsx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+            email: email,
+            password: values.password,
+            orgToken: token,
+          },
+        });
+
+        if (data) {
+          ToastAndroid.show('Successfully registered', ToastAndroid.LONG);
+          await AsyncStorage.setItem('org_token', data.createUser.token);
+          router.push('/dashboard/trainee');
```
Comment (ncutixavier) [inline app/auth/register.tsx]: After registering, why are you redirecting on the dashboard

Comment (fredshema) [reply app/auth/register.tsx]: Changed this to redirect to login

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**